### PR TITLE
Fix the bug in subsumption related to Global vars

### PR DESCRIPTION
**Pull request by rasoolmaghareh**

### CODE REVIEW

File: lib/Core/Executor.cpp [hunk 1]
```diff
@@ -1476,7 +1476,7 @@ Executor::StatePair Executor::branchFork(ExecutionState &current,
       // case also we extract the unsat core of the proof
       txTree->markPathCondition(current, unsatCore);
       if (WPInterpolant)
-        txTree->markInstruction(current.prevPC, true);
+        txTree->markInstruction(current.prevPC, false);
```
Comment (rasoolmaghareh) [inline lib/Core/Executor.cpp:1479]: May I ask why we have changed the second argument from false to true?

File: lib/Core/TxStore.cpp [hunk 2]
```diff
@@ -686,6 +686,17 @@ void TxStore::markFlow(ref<TxStateValue> target,
   }
 }
 
+// void TxStore::markGlobalVariables(ref<TxAllocationContext> ctx, ref<Expr>
```
Comment (rasoolmaghareh) [inline lib/Core/TxStore.cpp:689]: please remove these comments.

File: lib/Core/Executor.cpp (hunk 1 above)
Comment (xuanlinhha) [reply lib/Core/Executor.cpp:1479]: Because if res == Solver::True we mark 1, res == Solver::False we mark 2

**Comment by rasoolmaghareh** [comment]
@xuanlinhha I have checked this branch with the programs in the klee-examples\basic folder. There are a few issues:

1- In `addition_safe6.c`, the number of subsumptions is zero (can you please check the subsumptions manually).
2- `pointer_struct5.c`, `pointer_struct6.c` and `regexp_mark.c` crash.
3- `recursion2.c` takes a very long time (possibly the same issue as `addition_safe6.c`.

**Comment by rasoolmaghareh** [comment]
Summary of our online meeting:

•If a global variable is updated, TxTreeNode should contain a copy of its value. 
•This way when generating an interpolant, we have the intermediate value of the global var.

Example:

```
int a = 5;
main() {
  if (b > 0)
      a = a + 2;
  else
      skip;
  …
  if (a >= 7){
      …
  } else {
     …
  }
}
```

Here, the node after `b>0` should contain the updated value `a = 7`.
While the node after `b<=0` should read the original value `a = 5`.

After the false branch of the first if statment, `a=5`, then `a>=7` is false and `a=5` would be stored as interpolant.

At the true branch, `a=7`, then `a<7` is false and `a=7` is retrieved from the parent node and stored as interpolant.

**Comment by xuanlinhha** [comment]
I made a commit to get global value from parent node if it exists, otherwise, get the current one

**Comment by xuanlinhha** [comment]
1- In addition_safe6.c & 3- recursion2.c => Fixed
2- pointer_struct5.c, pointer_struct6.c and regexp_mark.c crash.
=> I see the global is a pointer to a structure, so I need to see more

**Comment by xuanlinhha** [comment]
I decided to skip checking global variable if it is a pointer

**Comment by rasoolmaghareh** [comment]
Noted.

**Pull request merged**